src/pages/mainPage.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import SideNavbar from 'components/SideNavbar';
+import styled from 'styled-components';
+import { removeCookie } from 'components/Shared/Cookies';
+import { useNavigate } from 'react-router';
+
+const LoginBtn = styled.button`
+  background-color: black;
+  display: flex;
+  position: absoulte;
+  justify-content: center;
+  color: white;
+  width: 80px;
+  height: 32px;
+  margin-left: 1700px;
+  margin-top: -900px;
+  padding-top: 5px;
+`;
+
+const mainPage = () => {
+  const loginInfo = useSelector((state) => state.auth);
+  const navigate = useNavigate();
+  const [isLogBtn, setIsLogBtn] = useState(loginInfo.isLoggedIn);
+
+  const loginClick = () => {
+    navigate(`/auth`);
+  };
+
+  useEffect(() => {
+    setIsLogBtn(loginInfo.isLoggedIn);
+  }, [loginInfo.isLoggedIn]);
+
+  const logoutClick = () => {
+    removeCookie('refresh_token');
+    loginInfo.isLoggedIn = false;
+    setIsLogBtn(false);
+  };
+  return (
+    <div>
+      <SideNavbar></SideNavbar>
+      {isLogBtn ? (
+        <LoginBtn onClick={logoutClick}>로그아웃</LoginBtn>
+      ) : (
+        <LoginBtn onClick={loginClick}>로그인</LoginBtn>
+      )}
```

Comment:
저번에도 말했지만 왜 굳이 전역 상태값을 받아와서 로컬에 넣고 사용하는지 모르겠습니다...

```suggestion
      {loginInfo.isLoggedIn ? (
        <LoginBtn onClick={logoutClick}>로그아웃</LoginBtn>
      ) : (
        <LoginBtn onClick={loginClick}>로그인</LoginBtn>
      )}
```

Reply:
좀 제가 삽질 좀 시도해본겁니다,
저는 저런식으로 작성하면 로그아웃버튼이 바로 로그인 버튼으로 바뀔줄 알았는데 "새로고침을 해야된다"라는 것을 새로고침 안한 상태에서 바로 바꿀순 없진 않을까해서 생각하다가 로컬 변수랑 저런식으로 작성하니깐 바로 적용이 된다는 것을 알게됬슴다....

Reply:
18ce553